clients/android/NewsBlur/src/com/newsblur/util/FeedUtils.java

```diff
@@ -270,6 +292,38 @@ protected void onPostExecute(NewsBlurResponse result) {
         }.execute();
     }
 
+    public static void muteFeeds(final Context context, final List<String> feedIds) {
+        updateFeedActiveState(context, feedIds, false);
+    }
+
+    public static void unmuteFeeds(final Context context, final List<String> feedIds) {
+        updateFeedActiveState(context, feedIds, true);
+    }
+
+    private static void updateFeedActiveState(final Context context, final List<String> feedIds, final boolean active) {
+        new AsyncTask<Void, Void, NewsBlurResponse>() {
+            @Override
+            protected NewsBlurResponse doInBackground(Void... arg) {
+                APIManager apiManager = new APIManager(context);
```

Comment:
Generally speaking, we have been trying to contain non-readonly API actions to being performed by the SyncService.  This allows them to be executed in a predictable order, with correct retry logic, and without creating race conditions.  The framework for this is the ReadingAction class.  Doing this API call in an AsyncTask will probably cause crashes or data loss if gets called in an offline or network brownout state.